tests/samplers_tests/test_nsgaii.py

```diff
@@ -146,6 +148,151 @@ def constraints_func(trial: FrozenTrial) -> Sequence[float]:
         assert trial.system_attrs[_CONSTRAINTS_KEY] == (trial.number,)
 
 
+@pytest.mark.parametrize("direction1", [StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE])
+@pytest.mark.parametrize("direction2", [StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE])
+@pytest.mark.parametrize(
+    "constraints_list",
+    [
+        [[]],  # empty constraint
+        [[-float("inf")], [-1], [0]],  # 1d constraints
+        [
+            [c1, c2] for c1 in [-float("inf"), -1, 0] for c2 in [-float("inf"), -1, 0]
+        ],  # 2d-constraints
+    ],
+)
+def test_constrained_dominates_feasible_vs_feasible(
+    direction1: StudyDirection, direction2: StudyDirection, constraints_list: List[List[float]]
+) -> None:
+    directions = [direction1, direction2]
+    # Check all pairs of trials consisting of these values, i.e.,
+    # [-inf, -inf], [-inf, -1], [-inf, 1], [-inf, inf], [-1, -inf], ...
+    values_list = [
+        [x, y]
+        for x in [-float("inf"), -1, 1, float("inf")]
+        for y in [-float("inf"), -1, 1, float("inf")]
+    ]
+    values_constraints_list = [(vs, cs) for vs in values_list for cs in constraints_list]
+
+    # The results of _constrained_dominates match _dominates in all feasible cases.
+    for (values1, constraints1) in values_constraints_list:
+        for (values2, constraints2) in values_constraints_list:
+            t1 = _create_frozen_trial(0, values1, constraints1)
+            t2 = _create_frozen_trial(1, values2, constraints2)
+            assert _constrained_dominates(t1, t2, directions) == _dominates(t1, t2, directions)
```

Comment:
@c-bata mentioned that what this test case covers may not be clear.

Reply:
After I checked the documentation of `constraints_fn`, it is easy to read this test case. Now I think that this test case is enough simple. Let me resolve this comment.